controller_manager/src/controller_manager.cpp

```diff
@@ -1034,6 +1034,62 @@ controller_interface::return_type ControllerManager::switch_controller(
       controller.info.claimed_interfaces.clear();
     }
   }
+
+  // Reordering the controllers
+  /// @note The following conditions needs to be handled while ordering the controller list
+  /// 1. The controllers that do not use any state or command interfaces are updated first
+  /// 2. The controllers that use only the state system interfaces only are updated next
+  /// 3. The controllers that use any of an another controller's reference interface are updated
+  /// before the preceding controller
+  /// 4. The controllers that use the controller's estimated interfaces are updated after the
+  /// preceding controller
+  /// 5. The controllers that only use the system's command interfaces are updated last
+  /// 6. All inactive controllers go at the end of the list
+  auto sorting_lambda = [](ControllerSpec ctrl_a, ControllerSpec ctrl_b)
```

Comment:
this is very elegant *however* I think it may be even nicer if it was an old-fashioned function... you know that ones with names ;) 

if it was that way I'd open a good first issue to do some more unit testing on trickier sorting setups without having to use the entire framework (which is great you've done but one of those is enough)

Reply:
Thank you @bmagyar. Sure!. I like this idea. I can think a bit more or the placement of this logic and yes then we can do the unit testing without the whole setup. I'll try to do it sometime soon.

Reply:
for now I think if you simply refactored this to move it to the header file and the cpp we could go ahead with a merge

Reply:
You mean that we I just separate it to a different function and then use it here?. Sure. I can do it.

Reply:
Yes

Reply:
Done

Reply:
thank you!